personal_finances/user/user_auth.py

```diff
@@ -0,0 +1,153 @@
+import base64
+import json
+import logging
+import boto3
+import bcrypt
+import jwt
+import os
+from datetime import datetime, timedelta
+from typing import Any, Callable, Tuple
+from decorator import decorator
+from personal_finances.user.user_auth_exceptions import (
+    AuthorizationHeaderNotPresent,
+    AuthorizationHeaderEmptyContent,
+    InvalidAuthorizationHeader,
+    InvalidLambdaEventInput,
+    UserNotFound,
+    PasswordNotMatch,
+    InvalidUserPassword,
+    InvalidUserName,
+    ServerSecretNotFound,
+    InvalidDynamoResponse,
+    get_server_response_by_exception,
+)
+
+LOGGER = logging.getLogger(__name__)
+USER_ID_MIN_LEN = 6
+USER_PASSWORD_MIN_LEN = 6
+
+
+def _decode_basic_auth(auth_header: str) -> Tuple[str, str]:
+    try:
+        auth_value = auth_header.replace("Basic ", "")
+        decoded_bytes = base64.b64decode(auth_value)
+        decoded_str = decoded_bytes.decode("utf-8")
+        userId, password = decoded_str.split(":", 1)
+        return userId, password
+    except Exception:
+        raise InvalidAuthorizationHeader()
+
+
+def _get_auth_header(event: dict) -> Any:
+    headers = event.get("headers", None)
+    if headers is None:
+        raise InvalidLambdaEventInput()
+
+    auth_header = headers.get("Authorization", None)
+    if auth_header is None:
+        raise AuthorizationHeaderNotPresent()
+
+    if auth_header == "":
+        raise AuthorizationHeaderEmptyContent()
+
+    return auth_header
+
+
+def _get_user_password(userId: str) -> str:
+    dynamodb = boto3.resource("dynamodb")
+
+    user_table = dynamodb.Table(os.environ["INFINEAT_DYNAMODB_USER_TABLE_NAME"])
+    response = user_table.get_item(
+        Key={"userId": userId},
+    )
+    print(response)
+    if "Item" not in response:
+        raise UserNotFound(f"User not found: {userId}")
+
+    user = response["Item"]
+    if "password" not in user or "userId" not in user:
+        raise InvalidDynamoResponse
+
+    return str(user["password"])
+
+
+def _get_jwt_secret() -> str:
+    jwt_session = boto3.client("secretsmanager")
+
+    try:
+        get_secret_value_response = jwt_session.get_secret_value(
+            SecretId=os.environ["INFINEAT_JWT_SECRET_NAME"]
+        )
+    except Exception:
```

Comment:
Be careful when catching all exceptions in a single block, if the environment variable is not configured, this block of code could raise a `KeyError`, and that wouldn't mean `ServerSecretNotFound`